wind_up/main_analysis.py

```diff
@@ -365,78 +366,6 @@ def yaw_offset_results(
     return results
 
 
-def check_for_ops_curve_shift(
-    pre_df: pd.DataFrame,
-    post_df: pd.DataFrame,
-    *,
-    wtg_name: str,
-    scada_ws_col: str,
-    pw_col: str,
-    rpm_col: str,
-    pt_col: str,
-    cfg: WindUpConfig,
-    plot_cfg: PlotConfig,
-    sub_dir: str | None = None,
-) -> dict[str, float]:
-    results_dict = {
-        "powercurve_shift": np.nan,
-        "rpm_shift": np.nan,
-        "pitch_shift": np.nan,
-    }
-    # check if all required columns are present
-    required_cols = [scada_ws_col, pw_col, pt_col, rpm_col]
-    for req_col in required_cols:
-        if req_col not in pre_df.columns:
-            msg = f"check_for_ops_curve_shift {wtg_name} pre_df missing required column {req_col}"
-            result_manager.warning(msg)
-            return results_dict
-        if req_col not in post_df.columns:
-            msg = f"check_for_ops_curve_shift {wtg_name} post_df missing required column {req_col}"
-            result_manager.warning(msg)
-            return results_dict
-    pre_dropna_df = pre_df.dropna(subset=[scada_ws_col, pw_col, pt_col, rpm_col]).copy()
-    post_dropna_df = post_df.dropna(subset=[scada_ws_col, pw_col, pt_col, rpm_col]).copy()
```

Comment:
I am not completely sure but I think this dropna logic is missing in the new `ops_curve_shift.py`

Reply:
> I am not completely sure but I think this dropna logic is missing in the new `ops_curve_shift.py`

This logic is here now https://github.com/resgroup/wind-up/blob/173f074d25a61c633fd41f58a711d85142f22383/wind_up/ops_curve_shift.py#L71

I've added NaN handling to the tests too...

[NaN included in df](https://github.com/resgroup/wind-up/blob/173f074d25a61c633fd41f58a711d85142f22383/tests/test_ops_curve_shift.py#L23) and the [assertion in a test](https://github.com/resgroup/wind-up/blob/173f074d25a61c633fd41f58a711d85142f22383/tests/test_ops_curve_shift.py#L112)